src/tigerbeetle/main.zig

```diff
@@ -37,17 +37,16 @@ pub const std_options = .{
 };
 
 pub fn main() !void {
-    try SigIllHandler.register();
-
     // TODO(zig): Zig defaults to 16MB stack size on Linux, but not yet on mac as of 0.11.
     // Override it here, so it can have the same stack size. Trying to set `tigerbeetle.stack_size`
-    // in build.zig doesn't work.
-    if (builtin.target.os.tag == .macos) {
-        std.posix.setrlimit(std.posix.rlimit_resource.STACK, .{
-            .cur = 16 * 1024 * 1024,
-            .max = 16 * 1024 * 1024,
-        }) catch @panic("unable to adjust stack limit");
-    }
+    // in build.zig doesn't work. setrlimit with 16MB errors with EINVAL on macOS 13 as well.
+    // So instead, we spawn a thread with the desired stack size instead.
+    const thread = try std.Thread.spawn(.{ .stack_size = 16 * 1024 * 1024 }, main_real, .{});
+    thread.join();
```

Comment:
I guess the real fix would be to finally fix that tail recursion in prefetch :rofl:

Reply:
I think we already fixed it :thinking: 

https://github.com/tigerbeetle/tigerbeetle/pull/1054#discussion_r1430322968

Reply:
:thinking: Ok, let's leave it as is, but then check if we could just remove this code after the upgrade is done.

In my mental model,  we _shouldn't_ be using all that much of a stack, so the stack size _shouldn't_ matter, but maybe I am wrong!